### PR TITLE
[2D] Remove enable_2d_with_fsdp() API and make remove_enable_2d_with_fsdp private

### PR DESCRIPTION
**Pull request by wz337**
As we have our new 2D flow out, we want to remove `enable_2d_with_fsdp()`.
In addition, we change pre_dp_module_transform to private, as we may need to change the UX later on.

### CODE REVIEW

**Review by fegin — APPROVED** [review]

File: docs/source/distributed.tensor.parallel.rst [hunk 1]
```diff
@@ -70,12 +70,7 @@ since we are parallelizing on the head dim of the ``MultiheadAttention`` module.
...
-To integrate with ``FullyShardedDataParallel``,
-users just need to call the following API explicitly:
-
-
-.. currentmodule:: torch.distributed.tensor.parallel.fsdp
-.. autofunction::  enable_2d_with_fsdp
+To integrate with ``FullyShardedDataParallel``.
 
 
 To integrate with ``DistributedDataParallel``,
```
Comment (wanchaol) [inline docs/source/distributed.tensor.parallel.rst:76]: any reason we don't remove the DDP extension part yet?

**Review by wanchaol — APPROVED** [review]
cool! one question

File: docs/source/distributed.tensor.parallel.rst (hunk 1 above)
Comment (wz337) [reply docs/source/distributed.tensor.parallel.rst:76]: The DDP part is not ready yet, but I guess we can also remove it from the doc if we don't want to promote it anyway.

**Pull request closed without merging**